shared/src/commonMain/kotlin/io/customer/shared/tracking/model/Activity.kt

```diff
@@ -0,0 +1,116 @@
+package io.customer.shared.tracking.model
+
+import io.customer.shared.common.CustomAttributes
+import io.customer.shared.tracking.api.model.Device
+import io.customer.shared.tracking.constant.ActivityType
+import io.customer.shared.tracking.constant.MetricEvent
+import kotlinx.serialization.SerialName
+
+/**
+ * Activity class to hold events for tracking. The class is sealed so only defined events can be
+ * tracked. Any new type supported should be added to this class.
+ *
+ * All inherited class should define their type in [SerialName] annotation to override type in
+ * serialization. See polymorphism in serialization for more details.
+ *
+ * @see [polymorphism in serialization][https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/polymorphism.md#custom-subclass-serial-name]
+ */
+@kotlinx.serialization.Serializable
+internal sealed interface Activity {
+    @SerialName("timestamp")
+    val timestamp: Long?
+
+    @SerialName("attributes")
+    val attributes: CustomAttributes?
+
+    /**
+     * Model version can help parsing json safely when migrating to newer versions of Activity models
+     */
+    @SerialName("modelVersion")
+    val modelVersion: Long
```

Comment:
This is helpful in parsing back as we are planning to put its json string in Database column and avoid adding too much tables for this.